### DISCUSSION TITLE
Backport PR #3029 to fix Nginx vulns flagged by Snyk

### DISCUSSION THREAD

**Discussion by strugee** [Mailu/Mailu#3182] [General]
Background: I want to independently monitor the security of Mailu's images so that when I run it in production, I can be alerted in a timely way and submit a PR, because Mailu's [security policy](https://github.com/Mailu/Mailu/security/policy) doesn't really specify what Mailu maintainers mean by "support". If this means that there's active monitoring of all image dependencies I wouldn't feel as much of a need to do this myself and would be willing to chip in upstream, but it isn't specified explicitly by the policy so I can't actually tell, and I don't want to assume because in my experience it's really common for projects shipping Docker images to not update their images with upstream distribution security patches.

Ask: please backport PR #3029, which bumps the version of Nginx being used in the Docker image. Snyk flags a _huge_ number of vulnerabilities, most of which appear to be in stuff that isn't actually exposed on the network (for example there's a bunch of curl issues and CVE-2023-4863, that really bad libwebp RcE from last fall, but I doubt any of that is actually network-exposed). However, there is one vulnerability, [CVE-2023-44487](https://www.cve.org/CVERecord?id=CVE-2023-44487), in nghttp2 which presumably _is_ actually used by Nginx. Presumably bumping the Nginx version is a pretty low-risk change, especially when the current version has had months of testing on `master`.

I'd be happy to prepare a PR for this, but I'm not sure exactly what the procedure is because it [seems like](https://mailu.io/2.0/contributors/workflow.html#pr-workflow) backports are automated _before_ PRs are merged.

**Comment by nextgens** [comment]
Hmm, that won't work; we can't upgrade to different major versions without testing.

What we would accept is a PR to move 2.0 from alpine 3.18.4 to 3.18.6

CVE-2023-44487 is addressed in https://git.alpinelinux.org/aports/commit/?id=908725a6208cfc88e76f208dc241b20d8d922fdd
According to https://gitlab.alpinelinux.org/alpine/aports/-/issues/15424 the version string was not patched and that's probably why your tool is having a problem with it; I am unclear on whether it is in 3.18.6. In any case we are talking about a DoS, nothing to write home about.
